lib/postgrex/protocol.ex

```diff
@@ -117,14 +134,27 @@ defmodule Postgrex.Protocol do
       types_lock: nil,
       prepare: prepare,
       messages: [],
-      ssl: ssl?,
+      ssl: ssl_opts,
       target_server_type: target_server_type,
       search_path: opts[:search_path]
     }
 
     connect_endpoints(endpoints, sock_opts ++ @sock_opts, connect_timeout, s, status, [])
   end
 
+  defp default_ssl_opts(opts) do
+    case Keyword.fetch(opts, :hostname) do
```

Comment:
If hostname is not set and PGHOST is used instead, the `server_name_indication` won't reflect the actual hostname, and server name verification will fail (as the TLS handshake is started on an established TCP connection and `:ssl` does not know the hostname that was used to establish the connection in the first place).

So either the docs would have to state that `hostname` option is mandatory when using `verify_peer`, or the logic to set `server_name_indication` would have to move to a later place where the actual hostname is known, no?

Reply:
I think this is good. PGHOST is read on `Postgrex.start_link` and set as the `opts[:hostname]`. If endpoints is set, that wins over hostname, but we will still use the hostname for SNI. I will update the docs!

Reply:
So if `hostname` is never supposed to be absent, why do we need the `case` statement?

Reply:
@voltone do you know if :server_name_indicator has any effect if we have `verify: :verify_none`?